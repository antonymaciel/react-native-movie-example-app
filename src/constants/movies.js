export const MOVIES_SUCCESS = 'MOVIES_SUCCESS';
export const MOVIES_FAILD = 'MOVIES_FAILD';
export const MOVIES_LOADING = 'MOVIES_LOADING';
export const MOVIES_CLEAN = 'MOVIES_CLEAN';