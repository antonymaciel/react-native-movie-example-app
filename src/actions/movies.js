import Api from "../api/apiMovies"
import { MOVIES_SUCCESS, MOVIES_FAILD, MOVIES_LOADING } from '../constants/movies'

export const moviesSuccess = (movies) => ({
    type: MOVIES_SUCCESS,
    movies 
});

export const moviesFaild = () => ({
    type: MOVIES_FAILD, 
});

export const moviesLoading = () => ({
    type: MOVIES_LOADING,
});

export const getMovies = (page) => (dispatch) => {
    dispatch(moviesLoading());
    Api.getMovies(page)
    .then((json) => {
        dispatch(moviesSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(moviesFaild());
    })
  };
 