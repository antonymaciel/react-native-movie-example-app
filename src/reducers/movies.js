import { MOVIES_SUCCESS, MOVIES_FAILD, MOVIES_LOADING } from '../constants/movies'

const initialState = {
    movies: null,
    loading: false,
    error: false
}


const movies = (state = initialState, action) => {
    switch(action.type){
        case MOVIES_SUCCESS:
            console.log(action);
            const movies = action.movies.results;
            return {
                movies: state.movies ? [ ...state.movies, ...movies] : movies ,
                loading: false, 
                error: false
            };
        case MOVIES_LOADING:
            console.log(action);
            return { 
                ...state, 
                loading: true,
                error: false
            };
        case MOVIES_FAILD:
            console.log(action);
            return { 
                ...state, 
                loading: false, 
                error: true
            };
        default:
                return state;
    } 
}

export default movies;