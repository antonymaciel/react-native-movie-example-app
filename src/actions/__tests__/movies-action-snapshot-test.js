import mockStore from 'redux-mock-store';
import { getMovies } from '../movies';
jest.mock('../../api/apiMovies');

it('should handle movies success action', async () => {
    const store = mockStore();
    await store.dispatch(getMovies(1));
    expect(store.getActions()).toMatchSnapshot();
});

