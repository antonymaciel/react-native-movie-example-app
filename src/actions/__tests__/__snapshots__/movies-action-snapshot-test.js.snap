// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle movies success action 1`] = `
Array [
  Object {
    "type": "MOVIES_LOADING",
  },
  Object {
    "movies": Array [
      Object {
        "title": "XMan",
      },
      Object {
        "title": "Spiderman",
      },
    ],
    "type": "MOVIES_SUCCESS",
  },
]
`;
